pallets/settlement/src/lib.rs

```diff
@@ -998,10 +1029,10 @@ impl<T: Trait> Module<T> {
 
     fn ensure_instruction_validity(instruction_id: u64) -> DispatchResult {
         let details = Self::instruction_details(instruction_id);
-        ensure!(
-            details.status == InstructionStatus::Pending,
-            Error::<T>::InstructionNotPending
-        );
+        // ensure!(
+        //     details.status == InstructionStatus::Pending,
```

Comment:
Presumably what we want is `!= InstructionStatus::Unknown`?

Reply:
The instruction not being a failed state is important and so accepting both pending and failed isn't a clean change. Might need to be a param for expected state of the instruction.

Reply:
I see. Is this code going to be commented in?